include/enums.h

```diff
@@ -298,7 +298,7 @@ enum eObstacle
     NUM_OBSTACLES // always last
 };
 
-enum eEquipment
+enum class eEquipment
```

Comment:
Not sure of the benefit of this eEquipment work - turning eqp into a std::map and implementing operator++ and -- just seems to me to be more code that doesn't make it better.

Reply:
#### TL;DR; I'll likely abandon this PR.

The `enum class` change was an effort to track all the paths that some value took before ending up as an index into an array somewhere, but it got hairy **real fast** as I started following those variables around.

The motivation behind this PR stemmed from seeing the wrong values being passed into functions that expected the values to be array indices.

For example, the 2nd parameter of `KMagic::res_throw()` is an index into the `KFighter::res[]` array, but:
```c++
src/eskill.cpp:    if (Magic.res_throw(target_fighter_index, S_STOP) == 0 && ...)
src/eskill.cpp:    if (Magic.res_throw(target_fighter_index, S_CHARM) == 0 && ...)
```
got me worried that we're likely setting some variable to a non-`eResistance` or non-`eWeapon` value, and as that variable was passed through multiple functions, the variable name kept changing so we never knew where it ended up.

I believe issue #124 describes what I was seeing: we were passing around some pretty inconsistent values, which may be what was causing bugs like #109, as well as ones that haven't been reported yet.

But you're right: It required all sorts of plumbing changes just so I could ensure the consistency of the data.

I'm fine abandoning this PR, but I'd definitely like to know a good way to track these items around.

Reply:
It's a tricky one this, specifically for eEquipment we sometimes want each slot (hand, head etc) to be a separate variable and sometimes we want to iterate through them like an array. I tried something like

```
union {
 uint8_t eqp[NUM_EQUIPMENT];
 struct {
  uint8_t eqp_weapon;
  uint8_t eqp_shield;
  ...
 };
};
```
but it didn't really help. My other idea was to have 
```
struct {
    uint8_t& operator[](eEquipment);
    uint8_t data[NUM_EQUIPMENT];
  } eqp;
```
where the operator[] would check the eEquipment before indexing but again it doesn't add much.

In general I find the widespread use of indexed arrays (e.g. `pidx`) to be a bit confusing and error-prone.

Reply:
Granted, this was a _long_ time ago, but I thought one of the reasons for the use of these indexed arrays was due to the lack of maps in C.

Do we want to look into adding another support library? I've used [better_enums](http://aantron.github.io/better-enums/) at two different jobs, which is pretty light-weight and has a few nice features we might want to consider:

1. Adds `.begin()` and `.end()` iterators for iteration.
2. Adds conversion to and from strings, which might be nice for the .mon data to have actual enum names instead of Magic Number values for every column.
3. Handles integer conversion and throws errors if we try to supply it with invalid values.